services/121-service/src/programs/fsp/africas-talking.service.ts

```diff
@@ -59,67 +62,22 @@ export class AfricasTalkingService {
       recipients: [],
     };
 
-    for (let item of paymentList) {
-      const recipient = {
-        phoneNumber: item.paymentAddress,
-        currencyCode: process.env.AFRICASTALKING_CURRENCY_CODE,
-        amount: amount,
-        metadata: {
-          programId: String(programId),
-          installment: String(installment),
-          did: String(item.did),
-          amount: String(amount),
-        },
-      };
-      payload.recipients.push(recipient);
-    }
+    const recipient = {
+      phoneNumber: paymentData.paymentAddress,
+      currencyCode: process.env.AFRICASTALKING_CURRENCY_CODE,
```

Comment:
Isn't the currency defined on the program?

So it would involve to query that data from here, based on the `programId`, or send only the value along all the way along the 'request-chain'...
I think it would depend on "how we handle other program-specific things" in these services to decide wich approach would be better scalable/reusable...